src/transforms/ampTransform.js

```diff
@@ -16,11 +16,15 @@
 
 const AmpOptimizer = require('@ampproject/toolbox-optimizer');
 const collectCss = require('../helpers/collectCss');
+const processOptions = require('../helpers/processOptions');
+
+const ampTransform = (eleventyConfig, providedOptions = {}) => {
+
+  const options = processOptions(providedOptions);
 
-const ampTransform = (eleventyConfig, options = {}) => {
   const ampOptimizer = createAmpOptimizer(options);
   eleventyConfig.addTransform('amp', async (content, outputPath) => {
-    if (!outputPath.endsWith('.html')) {
+    if (!outputPath.endsWith('.html') || !options.filter.test(outputPath)) {
```

Comment:
We could theoretically move the whole expression into a function on the options object:

```
if (options.isAmp(path)) {
   ...
}
```

Then we could remove all duplication between the two transforms and give users even more control over which files are processed.